src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,79 @@
+import { useCallback, useMemo, useState } from 'react';
+
+import debounce from 'lodash.debounce';
+
+import '../App.scss';
+
+import { Person } from '../types/Person';
+import { ListOfPeople } from './ListOfPeople';
+import { peopleFromServer } from '../data/people';
+
+interface Props{
+  delay: number,
+}
+
+const filterPeople = (people: Person[], currentQuery: string) => {
+  const query = currentQuery.toLowerCase();
+
+  return [...people].filter((person) => {
```

Comment:
`.filter` doesn't mutate the original array
```suggestion
  return people.filter((person) => {
```